### ISSUE TITLE
Adding media through the Media Library listing page doesn't work (wp-admin/upload.php)

### ISSUE THREAD

**Issue by liapotanin** [udx/wp-stateless#605]
The user of the plugin, received the message from the website form:

```
Adding media through the Media Library listing page doesn't work (wp-admin/upload.php). When I try to upload a new image, I don't think wp-stateless intercepts it because I get this error:

Unable to create directory wp-content/uploads/2021/04. Is its parent directory writable by the server?

It only works in wp-admin/media-new.php.

Wondering if you guys have any fix around this or an updated version?
```

**Comment by BGroever** [comment]
I observe something similar, when I use (wp-admin/upload.php) it saves the image in the local file directory wp-content/uploads/, but with wp-admin/media-new.php, the file is uploaded to gcloud storage as expected

I am running on a kubernetes cluster, so the file system uploads to wp-content/uploads/ will be removed when the container changes, it would be nice to have a feature with disable saving to local storage, so that users don't loose data

**Comment by mikgross** [comment]
I am having the same issue on App Engine from `wp-admin/media-new.php` and `wp-admin/upload.php`.
I run wp-stateless with completely stateless settings and app engine compatibility configured.

I get this error message:
`Unable to create directory wp-content/uploads/2021/05. Is its parent directory writable by the server?`

Just so you know, this directory is already created (I did it to prevent those errors), still it does not work.

For a stateless plugin, it is a pretty big show stopper...

**Comment by acollinsbl** [comment]
I'm unable to add media via the featured image select (request made to /wp-admin/async-upload.php), but fine using wp-admin/media-new.php. Here's the error:

`Unable to create directory wp-content/uploads/2021/04. Is its parent directory writable by the server?`

This is with stateless enabled.

**Comment by bethios** [comment]
Having the same issue.  Uploading through custom meta fields works fine but in the media library or featured image I get the same error messages as folks above

**Comment by michaijtsma** [comment]
I have the same issue on Google Cloud Run. Works on the /wp-admin/media-new.php page though.

**Comment by vsychov** [comment]
I has same problem, then I found that it's because GCS bucket uniform access, #521

**Comment by planv** [comment]
The fix was released in version `3.3`
Closing the ticket.

**Comment by engvie** [comment]
I had the problem "Unable to create directory...". I found that the problem was with my docker image when I realised it worked with a "default" wordpress image.

The problem I had was that this option was "Off" in the php.ini settings. This must be on.
`allow_url_fopen`

I hope this helps someone
